### PR TITLE
upgraded to hashicorp-vault-plugin v2.3

### PR DESCRIPTION
**Pull request by rcomanne**
Updated to hashicorp-vault-plugin v2.3

### CODE REVIEW

**Comment by rcomanne** [comment]
Needs more testing

File: src/main/java/io/jenkins/plugins/vault/VaultReadStep.java [hunk 1]
```diff
@@ -44,81 +53,96 @@ public void setKey(String key) {
...
+                .getVaultCredentialId() : Util.replaceMacro(step.credentialsId, environment);
+            String vaultUrl = step.vaultUrl == null || step.vaultUrl.isEmpty() ? vaultConfig.getConfiguration()
+                .getVaultUrl() : Util.replaceMacro(step.vaultUrl, environment);
 
-            listener.getLogger().append(String.format("using vault credentials \"%s\" and url \"%s\"", credentialsId, vaultUrl));
+            listener.getLogger()
+                .append(String.format("using vault credentials \"%s\" and url \"%s\"", credentialsId, vaultUrl));
 
             VaultAccessor vaultAccessor = new VaultAccessor();
-            vaultAccessor.init(vaultUrl);
```
Comment (alexeyc-lohika-tix) [inline src/main/java/io/jenkins/plugins/vault/VaultReadStep.java]: in case of `credentials == null` the  `init` method wont be executed

**Comment by casz** [comment]
Hi @rcomanne
I think it would be better if we deprecate this plugin.

And move the functionality to the `HashiCorp` plugin.

File: README.md [hunk 2]
```diff
@@ -54,9 +54,9 @@ By default, the plugin does not hide any accidental printing of secret to consol
 pipeline {
     agent any
     environment {
-        SECRET1    = vault path: 'secrets', key: 'password1', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds'
-        SECRET2    = vault path: 'secrets', key: 'password2', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds'
-        NOT_SECRET = vault path: 'secrets', key: 'username', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds'
+        SECRET1    = vault path: 'secrets', key: 'password1', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds', engineVersion: 2
```
Comment (casz) [inline README.md]: Does not have to be mandatory.

The default should be publicly available on `VaultConfiguration.DescriptorImpl.DEFAULT_ENGINE_VERSION`

**Comment by rcomanne** [comment]
> Hi @rcomanne
> I think it would be better if we deprecate this plugin.
> 
> And move the functionality to the `HashiCorp Vault plugin`.
> There will be less breakage that way 😆

Would definitely be worth it. Will try to implement and create a PR.

**Comment by rcomanne** [comment]
@casz PR has been created on the hashicorp-vault-plugin

**Comment by philmadden83** [comment]
Closing due to duplication https://github.com/jenkinsci/hashicorp-vault-pipeline-plugin/pull/11 and agreeing with @casz. However, I will be merging the other PR

**Pull request closed without merging**